DNN Platform/DotNetNuke.Web.Mvc/Startup.cs

```diff
@@ -13,11 +13,12 @@ public class Startup : IDnnStartup
     {
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddSingleton(serviceProvider => ControllerBuilder.Current.GetControllerFactory());
+            services.AddSingleton<IControllerFactory, DefaultControllerFactory>();
             services.AddSingleton<MvcModuleControlFactory>();
 
-            IDependencyResolver resolver = new DnnMvcDependencyResolver(Globals.DependencyProvider, DependencyResolver.Current);
-            DependencyResolver.SetResolver(resolver);
+            services.AddWebApiControllers();
```

Comment:
Since the container is now stored on the Scope does this mean the controller lifecycle is managed correctly? In other words does the GC properly clean it up?

Reply:
yes, this SO https://stackoverflow.com/a/40845639/499220 explains how it works.

The memory leak was actually because the controllers where registered as transient and resolved from the root (un-scoped) container that was never disposed (apart from application shutdown).

Reply:
This makes a lot of sense, thanks for explaining 👍

Reply:
Awesome, this is a great discovery!